### DISCUSSION TITLE
Number of recurring jobs in OSS version

### DISCUSSION THREAD

**Discussion by prabhat-krai** [jobrunr/jobrunr#685] [General]
Hi,
I want to use the OSS version of JobRunr
The doc says that it supports
"Up to 100 recurring jobs (depends on DB)"

I tried to schedule 30,000 jobs to test and they got scheduled and jobs were getting created for each recurring job.

So, I want to understand this 100 recurring job constraint and depends on DB part as I could not find clarification on this elsewhere.

Thank you.
![Uploading Screenshot 2023-02-10 at 10.31.36 AM.png…]()

**Comment by rdehuyss** [comment]
Well, originally I never designed JobRunr to support so many recurring jobs. For my projects, I typically only needed 5 to 10.

Soon, I received complaints that JobRunr missed recurring jobs when they had a couple of 100 recurring jobs. The missed jobs might not happen on your local PC as it depends on the combination of a lot of different factors (latency between DB and master node, amount of jobs in jobs table, ... ).

So, while you could have everything working on your local pc, be aware that once in PRD circumstances are different and I cannot guarantee that all your recurring jobs will run.

**Reply by prabhat-krai** [reply]
@rdehuyss thanks for the response.

I had 2 follow up questions.
1. How was this 100 number reached? Using that method we can calculate how much our prod DB can handle as recurring jobs.
2. Is 100 the max number of concurrent jobs workers can pick up or the max jobs that will be in enqueued state at one point?

**Reply by rdehuyss** [reply]
@prabhat-krai : I'm developing and supporting JobRunr mostly in my free time, and a lot of issues have been created regarding the amount of recurring jobs. While I do provide support for the Pro version, I'm spending way too much time from my personal life on supporting users with the OSS version of JobRunr. 

A quick search in the Github issues shows this:
- https://github.com/jobrunr/jobrunr/issues/437
- https://github.com/jobrunr/jobrunr/issues/426

**Comment by amodolo** [comment]
Hi @rdehuyss,
 i'm using the OSS version and a tried to schedule 500 recurring jobs as a POC...they worked. So a wondered if the 100 recurring job cap is subjected to a license usage constraint or they will in the future. I mean, the princing page is saying that there is cap of 100, but for me it's not clear if it's a "hard" technical limit (because of instability, api restriction or whatever...) or is a "soft" limit (license, usage terms, ...)

Thx

**Comment by rdehuyss** [comment]
Well, I may add the restriction in the future as I was contacted last week in panic by a company that deployed JobRunr OSS in production and not all the recurring jobs ran because of the things mentioned above. Then they start to doubt JobRunr whereas it is clearly mentioned on the website. I hope this helps.

**Reply by amodolo** [reply]
Ok, so if i'm using JobRuns OSS in a commercial product that schedules more than 100 recurring job, **at the moment**, is something that can be done (a part from some potential issue explained above). Right?

**Reply by rdehuyss** [reply]
I don't understand your question.

**Reply by amodolo** [reply]
I want to say that if I use JobRunr OSS in a project that has 1000 recurring jobs scheduled, I'm not breaking any terms of usage. Isn't it?

**Reply by rdehuyss** [reply]
No, but it probably will bite you in the ass because JobRunr OSS is not made for this amount of recurring jobs. That's why I'm thinking of adding the restriction in the future.

Can I ask why a Pro license is not an option?

**Reply by amodolo** [reply]
Pro license could be an option in the future (but a can't take this decision for my company). At this time, we are actually developing the solution and evaluate possible strength and weakness

**Reply by rdehuyss** [reply]
Drop me a note if there is anything I can help with!

**Reply by amodolo** [reply]
Sure, i will. Thx for you availability and you impressive effort :)
